pkg/ingester/ingester_v2.go

```diff
@@ -1438,6 +1510,78 @@ func (i *Ingester) compactBlocks(ctx context.Context, force bool) {
 	})
 }
 
+func (i *Ingester) closeAndDeleteIdleUsers(ctx context.Context) error {
+	for _, userID := range i.getTSDBUsers() {
+		if ctx.Err() != nil {
+			return nil
+		}
+
+		i.closeAndDeleteUserTSDBIfIdle(userID)
+	}
+
+	return nil
+}
+
+func (i *Ingester) closeAndDeleteUserTSDBIfIdle(userID string) {
```

Comment:
I think would be great to have a couple of metrics (counters) on succeeded and failed closing idle TSDBs. WDYT?

Reply:
I've added metrics for number of idle-checks, and various check results. PTAL